### PR TITLE
feat(userService): set up user profile

### PR DESCRIPTION
**Pull request by nmn-singh**
This pull request introduces the userService backend service. Key updates include:

Service Setup: Established the basic structure for the userService.
Database Connection: Integrated a connection to the database for data persistence.
User Profile: Created the UserProfile entity and implemented UserProfileService for handling user profiles.

### CODE REVIEW

File: userService/pom.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.2.2</version>
+		<relativePath/> 
+	</parent>
+	<groupId>com.crudops</groupId>
```
Comment (Arshjeet2003) [inline userService/pom.xml]: Provide a proper group id, artifact id and description.

File: userService/src/main/java/com/crudops/springbootcrudexample/SpringbootCrudExampleApplication.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.crudops.springbootcrudexample;
```
Comment (Arshjeet2003) [inline userService/src/main/java/com/crudops/springbootcrudexample/SpringbootCrudExampleApplication.java]: Not a good package name. Remove crudops and springbootcrudexample instead use something like userService.

File: userService/src/main/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#spring:
```
Comment (Arshjeet2003) [inline userService/src/main/resources/application.yml]: There should be no code that is not being used. Why is this commented?

File: userService/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# All the data source related properties and application properties configs go here
+server.port=9191
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/user_profile
+spring.datasource.username = root
```
Comment (Arshjeet2003) [inline userService/src/main/resources/application.properties]: username and password should come from .env file. Also add .env file to gitignore if not added. We should not show sensitive data.

**Comment by Arshjeet2003** [comment]
commits should have name of service. For eg - feat(userService) - ...

**Review by himcoder0 — APPROVED** [review]

File: userService/pom.xml (hunk 1 above)
Comment (nmn-singh) [reply userService/pom.xml]: this has been fixed

File: userService/src/main/java/com/crudops/springbootcrudexample/SpringbootCrudExampleApplication.java (hunk 2 above)
Comment (nmn-singh) [reply userService/src/main/java/com/crudops/springbootcrudexample/SpringbootCrudExampleApplication.java]: this has been fixed

File: userService/src/main/resources/application.yml (hunk 3 above)
Comment (nmn-singh) [reply userService/src/main/resources/application.yml]: this has been fixed

File: userService/src/main/resources/application.properties (hunk 4 above)
Comment (nmn-singh) [reply userService/src/main/resources/application.properties]: this has been fixed

File: userService/src/main/java/com/userServiceCrudOps/springbootcrudeUserService/userServiceCrudApplication.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    public static void main(String[] args) {
+        
+        Dotenv dotenv = Dotenv.load();
+        
+        
+        System.setProperty("SERVER_PORT", dotenv.get("SERVER_PORT"));
+        System.setProperty("MYSQL_URL", dotenv.get("MYSQL_URL"));
+        System.setProperty("MYSQL_USERNAME", dotenv.get("MYSQL_USERNAME"));
+		System.setProperty("MYSQL_PASSWORD", dotenv.get("MYSQL_PASSWORD"));
```
Comment (Arshjeet2003) [inline userService/src/main/java/com/userServiceCrudOps/springbootcrudeUserService/userServiceCrudApplication.java:14]: Proper linting required. No extra spaces between two lines. You can use 1 extra space to distinguish different parts but avoid 2 or more spaces. Last System.setProperty() should be in same indentation as others.

File: userService/.env [hunk 6]
```diff
@@ -0,0 +1,4 @@
+SERVER_PORT=9191
```
Comment (Arshjeet2003) [inline userService/.env]: Don't push the .env file. Instead create another pull request stating things to set up in readme file. Example - You can add a section in readme file which mentions that to start the service create a .env file and add SERVER_PORT=9191, etc. Also create a readme file giving general information about your service. This readme file should be within your service not the main readme.

File: userService/pom.xml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+	<groupId>com.userServiceCrudOps</groupId>
+	<artifactId>springboot-crud-example</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>springboot-crud-example</name>
+	<description>This service is responsible for handling queries related userservice</description>
+	<properties>
+		<java.version>17</java.version>
+	</properties>
+	<dependencies>
+<dependency>
```
Comment (Arshjeet2003) [inline userService/pom.xml]: Proper indentation required here.

File: userService/pom.xml [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>3.2.2</version>
+		<relativePath/> 
+	</parent>
+	<groupId>com.userServiceCrudOps</groupId>
+	<artifactId>springboot-crud-example</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>springboot-crud-example</name>
```
Comment (Arshjeet2003) [inline userService/pom.xml]: Change name here as well.

**Comment by Arshjeet2003** [comment]
Please do add some screenshots or a small video showing service is working.

File: userService/pom.xml (hunk 1 above)
Comment (Arshjeet2003) [reply userService/pom.xml]: Just click on resolve conversation for whichever has been fixed.

File: userService/src/main/java/com/userServiceCrudOps/springbootcrudeUserService/service/UserProfileService.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserProfileService {
+
+    @Autowired
+    private UserProfileRepository repository;
```
Comment (durgendra-singh) [inline userService/src/main/java/com/userServiceCrudOps/springbootcrudeUserService/service/UserProfileService.java:10]: Constructor injection is expected. Class variables should preferably be initialized using a constructor.

Comment (Arshjeet2003) [reply userService/src/main/java/com/userServiceCrudOps/springbootcrudeUserService/service/UserProfileService.java:10]: Autowired does the same thing as well i think.

Comment (durgendra-singh) [reply userService/src/main/java/com/userServiceCrudOps/springbootcrudeUserService/service/UserProfileService.java:10]: Constructor injection is considered one of the best practices in Spring Boot development due to its advantages in promoting immutability, explicit dependencies, mandatory initialization, easier unit testing, better compatibility with frameworks, and reduced complexity.

**Comment by nmn-singh** [comment]
https://github.com/user-attachments/assets/456dbe7f-4c6b-4b11-91b6-683dea1b29ca



> Please do add some screenshots or a small video showing service is working.

**Comment by nmn-singh** [comment]
> Please do add some screenshots or a small video showing service is working.


https://github.com/user-attachments/assets/f547c932-fc6a-49d1-bac0-428a687f48a9

**Comment by Arshjeet2003** [comment]
It seems that last commit(5077654) removes .env file and fixes other issues stated but this is not the correct way as it would not align with the commit description. Instead you should edit the commit in which .env file was added by not adding it. Similarly edit other commits instead of fixing issues in last commit. Also I mentioned no need to add readme file in this pull request. Create another pull request for readme file. Also edit commit message name to feat(userService): ... by editing the commits using rebase.

**Pull request merged**